src/chunk.js

```diff
@@ -0,0 +1,18 @@
+'use strict'
+
+const isNumber = require('./isNumber')
+
+const chunk = (anArray, chunkSize) => {
+  if (!Array.isArray(anArray) || !isInteger(chunkSize)) return []
+  const chunks = []
+  for (let i = 0; i < anArray.length; i += chunkSize) {
+    chunks.push(anArray.slice(i, i + chunkSize))
+  }
+  return chunks
```

Comment:
we can improve performance of this by doing all logic in single iteration O(n) vs O(n log n):

```suggestion
  const result = []
  let chunk = []
  for (let i = 0; i < anArray.length; i++) {
    chunk.push(anArray[i])
    if (chunk.length === chunkSize) {
      result.push(chunk)
      chunk = []
    }
  }
  if (chunk.length > 0) {
    result.push(chunk)
  }
  return result
```

Reply:
really interesting to benchmark it, I think first version should be faster

Reply:
```
Chunking Algorithm Benchmark

================================================================================

Test: Array size 100, Chunk size 10, 10000 iterations
--------------------------------------------------------------------------------
Slice method:    5.03ms total, 0.0005ms per iteration
Manual method:   5.20ms total, 0.0005ms per iteration
Winner: Slice method (1.03x faster)
Correctness: ✓ Both produce identical results

Test: Array size 1,000, Chunk size 50, 5000 iterations
--------------------------------------------------------------------------------
Slice method:    4.28ms total, 0.0009ms per iteration
Manual method:   20.99ms total, 0.0042ms per iteration
Winner: Slice method (4.91x faster)
Correctness: ✓ Both produce identical results

Test: Array size 10,000, Chunk size 100, 1000 iterations
--------------------------------------------------------------------------------
Slice method:    6.09ms total, 0.0061ms per iteration
Manual method:   38.54ms total, 0.0385ms per iteration
Winner: Slice method (6.33x faster)
Correctness: ✓ Both produce identical results

Test: Array size 100,000, Chunk size 1000, 100 iterations
--------------------------------------------------------------------------------
Slice method:    4.62ms total, 0.0462ms per iteration
Manual method:   45.32ms total, 0.4532ms per iteration
Winner: Slice method (9.80x faster)
Correctness: ✓ Both produce identical results

Test: Array size 1,000,000, Chunk size 10000, 10 iterations
--------------------------------------------------------------------------------
Slice method:    7.94ms total, 0.7942ms per iteration
Manual method:   62.54ms total, 6.2544ms per iteration
Winner: Slice method (7.88x faster)
Correctness: ✓ Both produce identical results

================================================================================

Benchmark complete!
```

Reply:
let me be a nerd and say that if we replace `array.push` with initialization of an array of correct size, i.e. `Array(chunkSize)` and then set values when iterating, we avoid array memory allocations when doing `push` 🤓